### PR TITLE
support agg key

### PR DESCRIPTION
**Pull request by Thearas**
Change-Id: Ic4fe2f9d5a2ddb69f8e07f1b04a6bcb56e39cd06

### CODE REVIEW

**Review by zgxme — COMMENTED** [review]
rest LGTM

File: gendata/tables.go [hunk 1]
```diff
@@ -46,7 +45,7 @@ var (
 	tablesTmpl        = defaultTablesTmpl
 	DBMS              = utils.MYSQL
 	SupportedKeyTypes = []string{
-		// KeyTypeAggregate,
+		KeyTypeAggregate,
 		KeyTypeDuplicate,
```
Comment (zgxme) [inline gendata/tables.go]: move to `consts.go`, then we just change it here later

**Review by zgxme — COMMENTED** [review]
rest LGTM

File: gendata/fields.go [hunk 2]
```diff
@@ -9,7 +9,15 @@ import (
 	lua "github.com/yuin/gopher-lua"
 )
 
-var fieldsTmpl = mustParse("fields", "`{{.fname}}` {{.types}} {{.sign}} {{.null}} {{.keys}}")
+// We can not determine if the column has agg_type while generating fields.
+// so add a placeholder for agg_type and replace it when we know if the table key type is `AGGREGATE`.
+const aggTypePlaceholder = "/*agg_type_placeholder*/"
+
```
Comment (zgxme) [inline gendata/fields.go]: move to consts.go

File: gendata/gendata.go [hunk 3]
```diff
@@ -71,19 +72,34 @@ func reorderFieldStmts(tableStmt *tableStmt, fieldStmts []string) []string {
...
+	}
+
 	// append remain non-key cols at last
 	orderedFieldStmts = append(orderedFieldStmts, fieldStmtsCopy...)
 
 	return orderedFieldStmts
 }
 
+var aggTypes = []string{"MAX", "MIN", "REPLACE" /* do not support "SUM", because only numeric type can use it */}
+
```
Comment (zgxme) [inline gendata/gendata.go:99]: 1. move to consts.go, then we just change it here later
2. append `REPLACE_IF_NOT_NULL ` or not? 
> https://doris.apache.org/zh-CN/docs/data-table/data-model
3. support `SUM` later?

File: gendata/tables.go [hunk 4]
```diff
@@ -20,8 +20,7 @@ type Tables struct {
 }
 
 const (
-	// TODO: Support agg type for non-key cols, to fix error: AGG_KEYS table should not specify aggregate type for non-key column
-	// KeyTypeAggregate = "AGGREGATE"
+	KeyTypeAggregate = "AGGREGATE"
 	KeyTypeDuplicate = "DUPLICATE"
```
Comment (zgxme) [inline gendata/tables.go]: move to consts.go

Comment (zgxme) [reply gendata/tables.go]: Keeping constants centralized

File: gendata/gendata.go (hunk 3 above)
Comment (Thearas) [reply gendata/gendata.go:99]: We will impl 2, 3 later when have time.

Comment (zgxme) [reply gendata/gendata.go:99]: > We will impl 2, 3 later when have time.

ok, get.

**Review by zgxme — APPROVED** [review]
LGTM

**Pull request merged**